services/src/main/java/org/jboss/windup/web/services/model/MigrationProject.java

```diff
@@ -59,8 +58,17 @@
     @Column(name = "version")
     private int version;
 
-    @Column(length = 256, unique = true, nullable = false)
-    @Size(min = 1, max = 256)
+    /**
+     * The project is being created in the wizard and should not be treated as a ready project.
+     */
+    @Column(nullable = true)
+    @ColumnDefault("FALSE")
+    private boolean provisional = true;
+
+    @Column(length = 120, unique = false, nullable = false)
+    @Size(min = 3, max = 120)
+    @Pattern(message = "The name must be at least 3 letters long.",
+            regexp = "\\s*\\w[- \\w]+\\w\\s*") // Allow whitespace around, limit to A-Za-z0-9-, allow spaces inside, at least 3 characters after whitespace trimmed.
```

Comment:
"!" is not a letter. And I think in at some point the PR had " The name must consist of 3 to 100 letters or spaces", only after several rebases it got mixed.  And it would be caught by UI anyway.